Rocket.Chat/Managers/MessageTextCacheManager.swift

```diff
@@ -41,13 +40,19 @@ class MessageTextCacheManager {
             text.setFontColor(MessageTextFontAttributes.defaultFontColor)
         }
 
-        resultText = NSMutableAttributedString(attributedString: text.transformMarkdown())
-        resultText.trimCharacters(in: .whitespaces)
-        resultText.highlightMentions(for: message)
-        resultText.highlightChannels(for: message)
+        let mentions = Array(message.mentions.flatMap { $0.username })
+        let channels = Array(message.channels.flatMap { $0.name })
+        let username = AuthManager.currentUser()?.username
+        DispatchQueue.global(qos: .background).async {
+            let finalText = NSMutableAttributedString(attributedString: text.transformMarkdown())
+            finalText.trimCharacters(in: .whitespaces)
+            finalText.highlightMentions(mentions, username: username)
+            finalText.highlightChannels(channels)
+            self.cache.setObject(finalText, forKey: key)
+            asyncUpdate?()
```

Comment:
Could this call have the `DispatchQueue.main.async { }`? I think it makes sense, after we change it to `completion`.